impl/core/src/main/java/io/serverlessworkflow/impl/WorkflowApplication.java

```diff
@@ -226,10 +236,20 @@ public Builder withEventPublisher(EventPublisher eventPublisher) {
       return this;
     }
 
+    public Builder withSecretManager(SecretManager secretManager) {
+      this.secretManager = secretManager;
+      return this;
+    }
+
+    public Builder withConfigManager(ConfigManager configManager) {
+      this.configManager = configManager;
+      return this;
+    }
+
     public <T> Builder withAdditionalObject(
         String name, WorkflowAdditionalObject<T> additionalObject) {
       if (additionalObjects == null) {
-        additionalObjects = new ConcurrentHashMap<>();
+        additionalObjects = new HashMap<>();
```

Comment:
Unrelated, but this can be a regular hashmap, since syncrhonization is not required for unvariant hashmap like this one once the application is built